anaUltraScurve.py

```diff
@@ -175,17 +175,33 @@ def plotAllSCurvesOnCanvas(vfatHistos, vfatHistosPanPin2=None, obsName="scurves"
     chanMaskGroup.add_option("--highNoiseCut", type="float", dest="highNoiseCut", default=1.0,
                       help="Threshold for setting the HighNoise maskReason, if channel (scurve_sigma > highNoiseCut) then HighNoise is set",
                       metavar="highNoiseCut")
-    chanMaskGroup.add_option("--deadChanCutLow", type="float", dest="deadChanCutLow", default=4.14E-02,
+    chanMaskGroup.add_option("--deadChanCutLow", type="float", dest="deadChanCutLow",
                       help="If channel (deadChanCutLow < scurve_sigma < deadChanCutHigh) then DeadChannel is set",
                       metavar="deadChanCutLow")
-    chanMaskGroup.add_option("--deadChanCutHigh", type="float", dest="deadChanCutHigh", default=1.09E-01,
+    chanMaskGroup.add_option("--deadChanCutHigh", type="float", dest="deadChanCutHigh",
                       help="If channel (deadChanCutHigh < scurve_sigma < deadChanCutHigh) then DeadChannel is set",
                       metavar="deadChanCutHigh")
     parser.add_option_group(chanMaskGroup)
 
     parser.set_defaults(outfilename="SCurveFitData.root")
     (options, args) = parser.parse_args()
     
+    #Default isVFAT2=false, isVFAT3=true
+    isVFAT3 = (not options.isVFAT2)
+    
+    #if isVFAT2, deadChanCut are set to default for v2 electronics
+    if options.isVFAT2:
+        if options.deadChanCutLow is None:
```

Comment:
Just a thought, will this default to none if with `type="float"`? Or will it default to `0.0`?  I would suggest for each of these two parameters to have `default=None` for these two parameters at lines 178 & 181 respectively.

Reply:
Reading some examples from https://docs.python.org/3/library/optparse.html looks like when the option is not set, options.name is `none` independently on the type. Nevertheless, this is not explicitly stated in the text so I will follow your suggestion.

Reply:
Thanks for checking this! Good to know. Keep in mind on cc7 python version defaults to 2.7; I doubt there’s a difference but sometimes functionality changes. 

In either case I think we’ve resolved this nicely.